### PR TITLE
Add was liberty and db2 #279

### PR DESCRIPTION
**Pull request by btyoshikawaa**
Please review #279.

### CODE REVIEW

**Comment by ikeyat** [comment]
This includes some proprietary libraries (DB2 JDBC Driver).
It is necessary to confirm redistribution license.

File: terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
+/*
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationTest.java]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/redirect/RedirectTest.java [hunk 2]
```diff
@@ -33,6 +33,9 @@
 
     @Value("${app.redirect.allowed.externalUrl}")
     String redirectionAllowedExternalUrl;
+    
+    @Value("${app.redirect.pageTitle.404Error}")
+    String pageTitle404Error;
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/redirect/RedirectTest.java]: - [ ]  Is this related for this PR really? It looks like unrelated refactoring.

File: terasoluna-gfw-functionaltest-parent/pom.xml [hunk 3]
```diff
@@ -135,12 +140,13 @@
...
         <postgresql.version>9.4-1205-jdbc41</postgresql.version>
         <ojdbc7.version>12.1.0.2</ojdbc7.version>
+        <db2jcc4.version>4.19.26</db2jcc4.version>
         <org.codehaus.cargo.cargo-maven2-plugin.version>1.4.16</org.codehaus.cargo.cargo-maven2-plugin.version>
         <cargo.jvmargs>-Xms512m -Xmx1024m -Duser.language=ja -Duser.country=JP -Duser.timezone=Asia/Tokyo</cargo.jvmargs>
         <cargo.daemon.url>http://localhost:18000</cargo.daemon.url>
         <cargo.container.id>tomcat8x</cargo.container.id>
         <cargo.container.url>http://archive.apache.org/dist/tomcat/tomcat-8/v${tomcat8.version}/bin/apache-tomcat-${tomcat8.version}.zip</cargo.container.url>
-        <tomcat8.version>8.0.29</tomcat8.version>
+        <tomcat8.version>8.0.30</tomcat8.version>
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-parent/pom.xml]: - [ ]  Is this related for this PR really?

File: terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java [hunk 4]
```diff
@@ -32,6 +32,9 @@ protected void doFilterInternal(HttpServletRequest request,
             HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
 
         String path = request.getServletPath();
+        if (path == null || path.length() == 0) {
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/pom.xml [hunk 5]
```diff
@@ -65,6 +65,10 @@
         </dependency>
         <dependency>
             <groupId>org.terasoluna.gfw</groupId>
+            <artifactId>terasoluna-gfw-validator</artifactId>
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/pom.xml]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java [hunk 6]
```diff
@@ -32,6 +32,9 @@ protected void doFilterInternal(HttpServletRequest request,
             HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
 
         String path = request.getServletPath();
+        if (path == null || path.length() == 0) {
+            path = request.getPathInfo();
+        }
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java]: - [ ] Is this file for this PR? Workaround for Liberty Profile?

File: terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationController.java [hunk 7]
```diff
@@ -0,0 +1,164 @@
+/*
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationController.java]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationForm.java [hunk 8]
```diff
@@ -0,0 +1,103 @@
+/*
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationForm.java]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/src/main/resources/ValidationMessages.properties [hunk 9]
```diff
@@ -1,3 +1,7 @@
 # this file contains the default messages of JSR 303 bean validation for validations provided in the common library
 
-org.terasoluna.gfw.common.codelist.ExistInCodeList.message = Does not exist in {codeListId}
\ No newline at end of file
+org.terasoluna.gfw.common.codelist.ExistInCodeList.message = Does not exist in {codeListId}
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/resources/ValidationMessages.properties]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/logging/xTrackMDCPutFilter.jsp [hunk 10]
```diff
@@ -21,7 +21,7 @@
 	<c:if test="${!(empty xTrackMDC)}">
 	${f:h(attributeName)}：<div id="xTrackMDC">${f:h(xTrackMDC)}</div>
 	</c:if>
-	<c:if test="${checkResponseHeaderXTrack}">
+	<c:if test="${!(empty checkResponseHeaderXTrack)}">
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/logging/xTrackMDCPutFilter.jsp]: - [ ] Is this file for this PR?

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp [hunk 11]
```diff
@@ -2,76 +2,72 @@
 <h1 class="center-content">Person Search Form</h1>
 
 <form class="center-content">
-    <spring:nestedPath path="personSearchForm">
-        <form:input path="name" />
-        <form:button id="searchButton">Search</form:button>
-    </spring:nestedPath>
+	<spring:nestedPath path="personSearchForm">
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp]: - [ ] Is this file for this PR?

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndCriteriaQuery.jsp [hunk 12]
```diff
@@ -2,54 +2,53 @@
 <h1 class="center-content">Person Search Form</h1>
 
 <form class="center-content">
-    <spring:nestedPath path="personSearchForm">
-        <form:input path="name" />
-        <form:button id="searchButton">Search</form:button>
-    </spring:nestedPath>
+	<spring:nestedPath path="personSearchForm">
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndCriteriaQuery.jsp]: - [ ] Don't change unrelated area.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndQueryTmplAndCriteriaQuery.jsp [hunk 13]
```diff
@@ -2,56 +2,57 @@
 <h1 class="center-content">Person Search Form</h1>
 
 <form class="center-content">
-    <spring:nestedPath path="personSearchForm">
-        <form:input path="name" />
-        <form:button id="searchButton">Search</form:button>
-    </spring:nestedPath>
+	<spring:nestedPath path="personSearchForm">
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndQueryTmplAndCriteriaQuery.jsp]: - [ ] Don't change unrelated area.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchQueryTmplAndCriteriaQuery.jsp [hunk 14]
```diff
@@ -2,54 +2,53 @@
 <h1 class="center-content">Person Search Form</h1>
 
 <form class="center-content">
-    <spring:nestedPath path="personSearchForm">
-        <form:input path="name" />
-        <form:button id="searchButton">Search</form:button>
-    </spring:nestedPath>
+	<spring:nestedPath path="personSearchForm">
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchQueryTmplAndCriteriaQuery.jsp]: - [ ] Don't change unrelated area.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/redirect/login.jsp [hunk 15]
```diff
@@ -1,7 +1,7 @@
   <h2>Redirect</h2>
   <br>
   <form:form action="${pageContext.request.contextPath}/login" method="post">
-    <c:if test="${param.error}">
+    <c:if test="${!(empty param.error)}">
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/redirect/login.jsp]: - [ ] Is this file for this PR? If this change is for other issue (mis-usage of EL), please create other issue and fix it in other branch.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/redirect/withgoto/loginWithGoTo.jsp [hunk 16]
```diff
@@ -1,7 +1,7 @@
   <h2>Redirect</h2>
   <br>
   <form:form action="${pageContext.request.contextPath}/redirect/withgoto/login" method="post">
-    <c:if test="${param.error}">
+    <c:if test="${!(empty param.error)}">
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/redirect/withgoto/loginWithGoTo.jsp]: - [ ] Is this file for this PR? If this change is for other issue (mis-usage of EL), please create other issue and fix it in other branch.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/redirect/withwhitelist/loginWithWhiteList.jsp [hunk 17]
```diff
@@ -1,7 +1,7 @@
   <h2>Redirect</h2>
   <br>
   <form:form action="${pageContext.request.contextPath}/redirect/withwhitelist/login" method="post">
-    <c:if test="${param.error}">
+    <c:if test="${!(empty param.error)}">
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/redirect/withwhitelist/loginWithWhiteList.jsp]: - [ ] Is this file for this PR? If this change is for other issue (mis-usage of EL), please create other issue and fix it in other branch.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/bytemax.jsp [hunk 18]
```diff
@@ -0,0 +1,20 @@
+<h2>@ByteMax Test</h2>
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/bytemax.jsp]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/bytemin.jsp [hunk 19]
```diff
@@ -0,0 +1,18 @@
+<h2>@ByteMin Test</h2>
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/bytemin.jsp]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/compare.jsp [hunk 20]
```diff
@@ -0,0 +1,49 @@
+<h2>@Compare Test</h2>
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/compare.jsp]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/index.jsp [hunk 21]
```diff
@@ -0,0 +1,13 @@
+<h2>Validation Function Test</h2>
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/index.jsp]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/welcome/index.jsp [hunk 22]
```diff
@@ -60,6 +60,10 @@
 			<td><a id="EL" href="${pageContext.request.contextPath}/el">EL
 					Function Test</a></td>
 		</tr>
+        <tr>
```
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/welcome/index.jsp]: - [ ] Is this file for this PR? It looks like test implementations for validation libraries...

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/logging/xTrackMDCPutFilter.jsp (hunk 10 above)
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/logging/xTrackMDCPutFilter.jsp]: I agreed with your fixes because most web articles use empty for both null and empty checking.

File: terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationTest.java (hunk 1 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationTest.java]: Sorry, It's difference that may occur due to I had to rebase too much.
I Canceled the wrong rebase, and re-push only the necessary commits.
This difference does not occur.

File: terasoluna-gfw-functionaltest-parent/pom.xml (hunk 3 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-parent/pom.xml]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/pom.xml (hunk 5 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/pom.xml]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationController.java (hunk 7 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationController.java]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationForm.java (hunk 8 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/validation/ValidationForm.java]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/resources/ValidationMessages.properties (hunk 9 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/resources/ValidationMessages.properties]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp (hunk 11 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndCriteriaQuery.jsp (hunk 12 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndCriteriaQuery.jsp]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndQueryTmplAndCriteriaQuery.jsp (hunk 13 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndQueryTmplAndCriteriaQuery.jsp]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchQueryTmplAndCriteriaQuery.jsp (hunk 14 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchQueryTmplAndCriteriaQuery.jsp]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/bytemax.jsp (hunk 18 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/bytemax.jsp]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/bytemin.jsp (hunk 19 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/bytemin.jsp]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/compare.jsp (hunk 20 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/compare.jsp]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/index.jsp (hunk 21 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/validation/index.jsp]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/welcome/index.jsp (hunk 22 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/welcome/index.jsp]: @ikeyat 
 Sorry, It's difference that may occur due to I had to rebase too much.
 I Canceled the wrong rebase, and re-push only the necessary commits.
 This difference does not occur.

File: terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/redirect/RedirectTest.java (hunk 2 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/redirect/RedirectTest.java]: @ikeyat 
This is related this PR d9b7c75f59c2b97eaae68b7fc5d98dff9427d67c changes of application-env.properties.
I changed the error page to be expected, because Websphere wrap Unexpected Exceptions in ServletException unlike other servers.

File: terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java (hunk 6 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java]: @ikeyat 
This is related this PR.
In Websphere(Liberty Profile), `HttpServletRequest#getServletPath()` return empty string unlike other servers, `HttpServletRequest#getPathInfo()` return the path to be expected.

**Comment by btyoshikawaa** [comment]
@ikeyat 
>This includes some proprietary libraries (DB2 JDBC Driver).
>It is necessary to confirm redistribution license.

I confirmed following notice in jdbc files(`db2_db2driver_for_jdbc_sqlj.zip`) and IBM Information.
`db2jcc4.jar` is redistributable.

```
IBM Data Server Driver for JDBC and SQLJ V4.19 Redistributables:

The following list includes files that are provided to Licensee pursuant to the Redistributables section of the IBM International Program License Agreements License Information that applies to this Program:

db2jcc.jar
db2jcc4.jar
sqlj.zip
sqlj4.zip
```

[The License Information](http://www14.software.ibm.com/cgi-bin/weblap/lap.pl?la_formnum=&li_formnum=L-LKSA-94UU9J&title=IBM+Data+Server+Driver+for+JDBC+and+SQLJ+%28JCC+Driver%29&l=en)

```
Redistributables 



If the Program includes components that are Redistributable, they will be identified in the REDIST file that accompanies the Program. In addition to the license rights granted in the Agreement, Licensee may distribute the Redistributables subject to the following terms: 

1) Redistribution must be in object code form only and must conform to all directions, instruction and specifications in the Program's accompanying REDIST or documentation; 

2) If the Program's accompanying documentation expressly allows Licensee to modify the Redistributables, such modification must conform to all directions, instruction and specifications in that documentation and these modifications, if any, must be treated as Redistributables; 

3) Redistributables may be distributed only as part of Licensee's application that was developed using the Program ("Licensee's Application") and only to support Licensee's customers in connection with their use of Licensee's Application. Licensee's Application must constitute significant value add such that the Redistributables are not a substantial motivation for the acquisition by end users of Licensee's software product; 

4) If the Redistributables include a Java Runtime Environment, Licensee must also include other non-Java Redistributables with Licensee's Application, unless the Application is designed to run only on general computer devices (for example, laptops, desktops and servers) and not on handheld or other pervasive devices (i.e., devices that contain a microprocessor but do not have computing as their primary purpose); 

5) Licensee may not remove any copyright or notice files contained in the Redistributables; 

6) Licensee must hold IBM, its suppliers or distributors harmless from and against any claim arising out of the use or distribution of Licensee's Application; 

7) Licensee may not use the same path name as the original Redistributable files/modules; 

8) Licensee may not use IBM's, its suppliers or distributors names or trademarks in connection with the marketing of Licensee's Application without IBM's or that supplier's or distributor's prior written consent; 

9) IBM, its suppliers and distributors provide the Redistributables and related documentation without obligation of support and "AS IS", WITH NO WARRANTY OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING THE WARRANTY OF TITLE, NON-INFRINGEMENT OR NON-INTERFERENCE AND THE IMPLIED WARRANTIES AND CONDITIONS OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE; 

10) Licensee is responsible for all technical assistance for Licensee's Application and any modifications to the Redistributables; and 

11) Licensee's license agreement with the end user of Licensee's Application must notify the end user that the Redistributables or their modifications may not be i) used for any purpose other than to enable Licensee's Application, ii) copied (except for backup purposes), iii) further distributed or transferred without Licensee's Application or iv) reverse assembled, reverse compiled, or otherwise translated except as specifically permitted by law and without the possibility of a contractual waiver. Furthermore, Licensee's license agreement must be at least as protective of IBM as the terms of this Agreement. 
```

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp (hunk 11 above)
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp]: This change has not canceled yet. Please remove.
commit 6b164f44bb7d2bedb3a5e2051bda109fbb68501d is included in this PR.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndQueryTmplAndCriteriaQuery.jsp (hunk 13 above)
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndQueryTmplAndCriteriaQuery.jsp]: This change has not canceled yet. Please remove.
commit 6b164f44bb7d2bedb3a5e2051bda109fbb68501d is included in this PR.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchQueryTmplAndCriteriaQuery.jsp (hunk 14 above)
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchQueryTmplAndCriteriaQuery.jsp]: This change has not canceled yet. Please remove.
commit 6b164f44bb7d2bedb3a5e2051bda109fbb68501d is included in this PR.

File: terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java (hunk 6 above)
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java]: - [ ] I got it but please leave comments on the source code for this as following.

// Workaround for application servers that don't provide getServletPath(), eg. WebSphere Liberty Profile 8.5.

File: terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/redirect/RedirectTest.java (hunk 2 above)
Comment (ikeyat) [inline terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/redirect/RedirectTest.java]: @btyoshikawaa 
I think it does not make sense. This change is just removing hard coding and putting it on properties. It looks a refactoring and assertion is not changed completely.
I thinks it's OK for this PR contains tiny refactoring and I will merge, but I'm just afraid you have some mis-understandings.

**Comment by ikeyat** [comment]
@btyoshikawaa 
According to the License,
``> 5) Licensee may not remove any copyright or notice files contained in the Redistributables;``

Are there any license or copyright file in ``db2_db2driver_for_jdbc_sqlj.zip``?

File: terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java (hunk 6 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/java/org/terasoluna/gfw/functionaltest/app/exceptionhandling/ExceptionLevelResolverFilter.java]: @ikeyat 
Thanks, I'd added comment you presented.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp (hunk 11 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp]: @ikeyat 
6b164f44bb7d2bedb3a5e2051bda109fbb68501d is neccessary for this PR.
But It had been changed space to tab unintentionally by the eclipse auto format.
I changed tab to space in those files agein.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndCriteriaQuery.jsp (hunk 12 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndCriteriaQuery.jsp]: @ikeyat 
6b164f44bb7d2bedb3a5e2051bda109fbb68501d is neccessary for this PR.
But It had been changed space to tab unintentionally by the eclipse auto format.
I changed tab to space in those files agein.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndQueryTmplAndCriteriaQuery.jsp (hunk 13 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchPathTmplAndQueryTmplAndCriteriaQuery.jsp]: @ikeyat 
6b164f44bb7d2bedb3a5e2051bda109fbb68501d is neccessary for this PR.
But It had been changed space to tab unintentionally by the eclipse auto format.
I changed tab to space in those files agein.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchQueryTmplAndCriteriaQuery.jsp (hunk 14 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/searchQueryTmplAndCriteriaQuery.jsp]: @ikeyat 
6b164f44bb7d2bedb3a5e2051bda109fbb68501d is neccessary for this PR.
But It had been changed space to tab unintentionally by the eclipse auto format.
I changed tab to space in those files agein.

**Comment by btyoshikawaa** [comment]
@ikeyat 
I'd copied license files in `db2_db2driver_for_jdbc_sqlj.zip`.
Please show 80406abd12da3ba24d0eeab366731ef85aeea243.

File: terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp (hunk 11 above)
Comment (btyoshikawaa) [inline terasoluna-gfw-functionaltest-web/src/main/webapp/WEB-INF/views/pagination/search.jsp]: @ikeyat 
This changes to be expected to escape `ParseException` when new line exists in expression.
But I've to understand that there is a following of such a solution, I removed These unneccessary changes.

```xml
<jsp-attribute name="doNotEscapeWhitespaceCharsInExpression" value="true" />
```

**Comment by btyoshikawaa** [comment]
I had rebased master for resolving conflict.

**Comment by ikeyat** [comment]
@btyoshikawaa 
After consideration, I would like to remove jdbc jar in this branch
Jdbc jar in this proj is never used in actual test because jdbc jar is needed to be located in application server that produces datasource.
Please remove jdbc jar from this PR.

**Comment by btyoshikawaa** [comment]
@ikeyat 
I removed jdbc jar and license files.
But in initdb project, need to this jar file for executing sqls. I left comments on pre-installing jar in local ripository.
